python/lsst/daf/butler/registry/_registry.py

```diff
@@ -1227,8 +1227,12 @@ def queryDatasetTypes(self, expression: Any = ..., *, components: Optional[bool]
                 yield datasetType
                 if components and datasetType.isComposite():
                     # Automatically create the component dataset types
-                    for component in datasetType.makeAllComponentDatasetTypes():
-                        yield component
+                    try:
+                        componentsForDatasetType = datasetType.makeAllComponentDatasetTypes()
+                    except KeyError as err:
+                        _LOG.warn(f"Could not load storage class {err} for {datasetType.name}; "
+                                  "if it has components they will not be included in query results.")
+                    yield from componentsForDatasetType
```

Comment:
I assume if `makeAllComponentDatasetTypes` throws an exception, then `componentsForDatasetType` is `None`.  Is it okay to do `yield from None`?

Reply:
Good catch!  I think it's actually an undefined variable, so the `yield from` is definitely going to break and I need to move it into a `try/else`, probably.